### PR TITLE
Dogu Agent update

### PR DESCRIPTION
**Pull request by yowpark**
# PR Template

## Checklist

- [ ] E2E finished.
- [x] Local, Self Hosted and Development environment test finished.
- [x] Local, E2E, Self Hosted, Development and Production env variable checked.
- [ ] (Backend only) DB migration file creation checked.
- [ ] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

https://github.com/dogu-team/dogu/issues/165

- Console Frontend
  - Add Host Update button on each host menu
- Console Backend
  - add host-app.controller to trigger update
  - read package.json 's version at config.ts on boot time
- Database
  - add architecture column to host.entity. used to determine install file
- Host Agent
  - add update.processor to proceed update
    - macOS: Detach terminal to run update script
    - windows: Detach cmd to run .cmd script

### CODE REVIEW

**Comment by oneofthezombies** [comment]
thank you!

File: projects/console-web-front/src/components/hosts/HostListController.tsx [hunk 1]
```diff
@@ -153,6 +166,21 @@ const HostItem = ({ host }: HostItemProps) => {
...
+          modalTitle={t('device-farm:hostItemUpdateMenu')}
+          modalButtonTitle={t('device-farm:hostUpdateModalButtonText')}
+          modalContent={<StyledDeleteModalContent>{t('device-farm:hostUpdateModalContentInfo')}</StyledDeleteModalContent>}
+          confirmButtonId="host-update-confirm-btn"
+        >
+          {t('device-farm:hostItemUpdateMenu')}
+        </MenuItemButton>
+      ),
+      key: 'update',
+    },
```
Comment (Ikalli) [inline projects/console-web-front/src/components/hosts/HostListController.tsx:183]: Displaying this menu when the host is updatable could be more friendly..!

**Review by Ikalli — CHANGES_REQUESTED** [review]
Need to improve UX - disable or hide update host button if the host is latest version

File: projects/console-web-front/src/components/hosts/HostListController.tsx (hunk 1 above)
Comment (yowpark) [reply projects/console-web-front/src/components/hosts/HostListController.tsx:183]: There is api to get latest apps {{API_BASE_URL}}/downloads/dogu-agent/latest! https://github.com/dogu-team/dogu/pull/319/files#diff-3d51b5a5e5b7b359f659f7bd551616df75e3e82d289cecb10f0293a40e9f4b08

**Pull request merged**